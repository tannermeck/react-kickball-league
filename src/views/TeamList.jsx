function TeamList(){
    return (
        <>
            <h1>Hello from teams</h1>
        </>
    )
}
export default TeamList;