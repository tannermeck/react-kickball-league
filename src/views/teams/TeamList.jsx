import { useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import { getTeams } from '../../services/teams';
import './teamdetail.css';

function TeamList(){
    const [teams, setTeams] = useState([])
    const [loading, setLoading] = useState(true)

    useEffect(() => {
        async function getTeamsList(){
            const teamList = await getTeams();
            setTeams(teamList)
            setLoading(false)
        }
        getTeamsList()
    }, [])
    if (loading) return <h1>Loading Teams!..</h1>

    return (
        <>
            <h1 className='detailHeader'>Teams:</h1>
            <ul>
                {teams.map((team) => {
                    return (
                        <NavLink key={team.id} to={`/teams/${team.id}`}>
                            <li key={team.id}>{team.name}</li>
                        </NavLink>
                    )
                })}
            </ul>
        </>
    )
}
export default TeamList;