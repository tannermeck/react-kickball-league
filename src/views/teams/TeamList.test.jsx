import { screen, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import TeamList from './TeamList';

it('should render out a team list component', async () => { 
    const { container } = render(
                            <MemoryRouter>
                                <TeamList />
                            </MemoryRouter>)
    await screen.findByText(/Loading.../i)
    const label = await screen.findByText('Teams:', {exact: false})
    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})