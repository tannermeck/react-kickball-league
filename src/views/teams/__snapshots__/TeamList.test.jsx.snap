// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render out a team list component 1`] = `
<div>
  <h1
    class="detailHeader"
  >
    Teams:
  </h1>
  <a
    class="newTeamLink"
    href="/teams/new"
  >
    Add New Team
  </a>
  <ul>
    <div
      class="teamListContainer"
    >
      <a
        href="/teams/1"
      >
        <li>
          Bridge City Sneakers
        </li>
      </a>
      <button
        class="deleteButton"
        type="button"
      >
        Delete Bridge City Sneakers
      </button>
      <a
        class="editTeam"
        href="/teams/edit/1"
      >
        Edit Team
      </a>
    </div>
    <div
      class="teamListContainer"
    >
      <a
        href="/teams/15"
      >
        <li>
          Jail Blazers
        </li>
      </a>
      <button
        class="deleteButton"
        type="button"
      >
        Delete Jail Blazers
      </button>
      <a
        class="editTeam"
        href="/teams/edit/15"
      >
        Edit Team
      </a>
    </div>
    <div
      class="teamListContainer"
    >
      <a
        href="/teams/17"
      >
        <li>
          Peter Pipers
        </li>
      </a>
      <button
        class="deleteButton"
        type="button"
      >
        Delete Peter Pipers
      </button>
      <a
        class="editTeam"
        href="/teams/edit/17"
      >
        Edit Team
      </a>
    </div>
  </ul>
</div>
`;
